Sources/Amplitude/DefaultTrackingOptions.swift

```diff
@@ -8,20 +9,50 @@ public class DefaultTrackingOptions {
         DefaultTrackingOptions(sessions: false, appLifecycles: false, screenViews: false)
     }
 
-    public var sessions: Bool = true
-    public var appLifecycles: Bool
-    public var screenViews: Bool
-    public var userInteractions: Bool
+    public var sessions: Bool {
+        didSet {
+            if sessions {
+                autocapture?.insert(.sessions)
+            } else {
+                autocapture?.remove(.sessions)
+            }
+        }
+    }
+
+    public var appLifecycles: Bool {
+        didSet {
+            if appLifecycles {
+                autocapture?.insert(.appLifecycles)
+            } else {
+                autocapture?.remove(.appLifecycles)
+            }
+        }
+    }
+
+    public var screenViews: Bool {
+        didSet {
+            if screenViews {
+                autocapture?.insert(.screenViews)
+            } else {
+                autocapture?.remove(.screenViews)
+            }
+        }
+    }
 
     public init(
         sessions: Bool = true,
         appLifecycles: Bool = false,
-        screenViews: Bool = false,
-        userInteractions: Bool = false
+        screenViews: Bool = false
     ) {
         self.sessions = sessions
         self.appLifecycles = appLifecycles
         self.screenViews = screenViews
-        self.userInteractions = userInteractions
+    }
+
+    private var autocapture: AutocaptureOptions?
```

Comment:
this is a struct, which is a value type. Changes here won't be reflected outside of this var - was this intended to change the amplitude object's autocapture options as well?

Reply:
This was an error and the behavior have been broken here (we were not actually tracking changes to the options as the autocapture property is a struct).